test/framework/filetools.py

```diff
@@ -1212,6 +1233,15 @@ def test_apply_regex_substitutions(self):
         path = os.path.join(self.test_prefix, 'nosuchfile.txt')
         self.assertErrorRegex(EasyBuildError, error_pat, ft.apply_regex_substitutions, path, regex_subs)
 
+        # make sure apply_regex_substitutions can patch files that include UTF-8 characters
+        testtxt = b"foo \xe2\x80\x93 bar"  # This is an UTF-8 "-"
+        ft.write_file(testfile, testtxt)
+        ft.apply_regex_substitutions(testfile, [('foo', 'FOO')])
+        txt = ft.read_file(testfile)
+        if sys.version_info[0] == 3:
+            testtxt = testtxt.decode('utf-8')
```

Comment:
Hmm, shouldn't this `decode` be done in `read_file` itself, to maintain backward compatibility?

Reply:
It does. This is only the decode on the  `testtxt` not `txt`

Reply:
ok, gotcha